src/cubic_loader/qlik/ods_qlik.py

```diff
@@ -0,0 +1,402 @@
+        new_last_schema = self.etl_status.last_schema
+        if last_schema is not None:
+            new_last_schema = last_schema
+
+        self.etl_status = TableStatus(
+            db_fact_table=self.db_fact_table,
+            current_snapshot_ts=new_shapshot_ts,
+            last_cdc_ts=new_cdc_ts,
+            last_schema=new_last_schema,
+        )
+
+    def load_etl_status(self) -> TableStatus:
+        """pull and verify table status from file"""
+        if s3_object_exists(self.status_path):
+            last_status = json.load(s3_get_object(self.status_path))
+            return_satus = TableStatus(
+                db_fact_table=self.db_fact_table,
+                current_snapshot_ts=last_status["current_snapshot_ts"],
+                last_cdc_ts=last_status["last_cdc_ts"],
+                last_schema=last_status["last_schema"],
+            )
+        else:
+            return_satus = TableStatus(
+                db_fact_table=self.db_fact_table,
+                current_snapshot_ts=self.last_s3_snapshot_dfm.ts,
+                last_cdc_ts="",
+                last_schema=dfm_schema_to_json(self.last_s3_snapshot_dfm),
+            )
+            self.save_status(return_satus)
+
+        return return_satus
+
+    def save_status(self, status: TableStatus) -> None:
+        """write status to file path"""
+        with NamedTemporaryFile(mode="w+") as f:
+            json.dump(status._asdict(), f)
+            f.flush()
+            s3_upload_file(f.name, self.status_path)
+
+    def rds_snapshot_load(self) -> None:
+        """Perform load of initial load files to history table"""
+        self.db.execute(create_history_table_partitions(self.db_history_table, self.last_s3_snapshot_dfm.ts))
+
+        bucket, prefix = s3_split_object_path(self.last_s3_snapshot_dfm.path.rsplit("/", maxsplit=1)[0])
+        for s3_path in s3_list_objects(bucket, prefix, in_filter=".csv.gz"):
+            remote_csv_gz_copy(s3_path, f"{ODS_SCHEMA}.{self.db_load_table}")
+
+        load_update = (
+            f"UPDATE {ODS_SCHEMA}.{self.db_load_table} "
+            f"SET header__timestamp=to_timestamp('{self.last_s3_snapshot_dfm.ts}','YYYYMMDDTHHMISSZ') "
+            f", header__change_oper='L' "
+            f", header__change_seq='SNAPSHOT_LOAD'"
+            f"WHERE header__timestamp IS NULL;"
+        )
+        self.db.execute(load_update)
+        table_copy = (
+            f"INSERT INTO {ODS_SCHEMA}.{self.db_history_table} " f"SELECT * FROM {ODS_SCHEMA}.{self.db_load_table};"
+        )
+        self.db.execute(table_copy)
+        self.db.truncate_table(f"{ODS_SCHEMA}.{self.db_load_table}")
+        self.db.vaccuum_analyze(f"{ODS_SCHEMA}.{self.db_history_table}")
+
+        self.update_status(
+            self.etl_status.current_snapshot_ts,
+            "0",
+            self.etl_status.last_schema,
+        )
+
+    def rds_cdc_load(self) -> None:
+        """Perform load of CDC files to history table"""
```

Comment:
Loading CDC files is the most complicated part of this process. 

For each table that is loaded a schema comparison is made against the expected table schema from the ETL status file. If a new column is detected, the initial load operation is aborted and then re-tried after the new column have been added to the RDS tables. 

These load operations are threaded, because they are performed with a `psql` sub-process command, that utilizes very little compute resources.